### ISSUE TITLE
Can create multiple usergroups with the same name

### ISSUE THREAD

**Issue by SvePu** [mybb/mybb#2691]
I'd consider this a bug so I'll post it here and see if it effects anyone else. I can create many user groups with the exact same group name. There should be some sort of check.

Original thread: https://community.mybb.com/thread-207529.html

**Comment by SvePu** [comment]
Please review #2692

**Comment by Shade-** [comment]
The discussion is not over. Next time, please wait for a staff member to push the issue to GitHub once there is an agreement on how to proceed with the issue, thank you.

**Comment by Ben-MyBB** [comment]
@mybb/developers @mybb/management @mybb/sqa @mybb/community-team What does everyone think about this? I think it's a bug IMO.

**Comment by kawaii** [comment]
Personally, I wouldn't say it crosses the threshold to be classed as a bug. It's not like usernames where there's the importance of not having a direct collision, since there's nowhere in MyBB where you actually need to type a group name to perform an action. Ultimately it's a question of standards - since I can't see this 'bug' actually breaking anything. :smile_cat:

**Comment by Ben-MyBB** [comment]
Well users could have multiple of the same groups and not realise, it's odd to allow people to have multiple groups with the same name. That's like allowing multiple users with the same username.

**Comment by WildcardSearch** [comment]
This isn't a bug. The **id**entity is in tact. Creating two groups with the same name might be desired if the admin wants to create two different groups with slightly different permission, but are indistinguishable to regular users.

**Comment by Cu8eeeR** [comment]
There should be a notice that the usergroup will have same name,,, at least. I think it is a bug but many admins use this as a feature

**Comment by kawaii** [comment]
I agree with @Cu8eeeR in this case. We should simply notify the user that a group with the same name already exists and ask if they wish to continue. Perhaps @SvePu can modify his pull request to accommodate this? :)

**Comment by Its-Nasyr** [comment]
I don't believe this to be a bug, I agree with Wildcard on this matter. Some communities have two variants of Administrators which leads them to creating multiple groups with slightly different names.

**Comment by wagey** [comment]
Don't see the benefit in changing this, period. You can always distinguish a set of groups that share the same name by just looking at the gid.

**Comment by SvePu** [comment]
@wagey -  In groupselect settings options you can't see the gids of groups.

**Comment by kawaii** [comment]
How about we simply display GIDs in the ACP then so that novice users can quickly check if they don't understand that the URI contains the GID?

**Comment by WildcardSearch** [comment]
@Cu8eeeR I see your point with the warning for inexperienced users, but I still don't think we need to prevent admin from creating two groups with the same name. Perhaps we can slightly alter the `flash_message` for saving a user group with the same name as an existing user group.

@mybb/developers I suggest that when admin creates a new user group, or edits an existing group, using a title that is already used by an existing user group, the process be allowed to complete, but a warning is flashed to admin:

```
This user group has the same title as an existing user group.
```

**Comment by Shade-** [comment]
I agree with @WildcardSearch's solution.

**Comment by Its-Nasyr** [comment]
@WildcardSearch I like your idea for a quick warning to members. Makes things much easier and gives a quick resolve but still letting administrators have more flexibility.

**Comment by JordanMussi** [comment]
The name of a group isn't required to be unique so we shouldn't force them to be unique. It is up to the user if they want to distinguish between group names.

Anyway, once a group is saved the user is redirected to the groups list where it will be obvious if they have multiple groups with the same name so the message is redundant. Having a message could be confusing to some users as they may think they've done something wrong.

Therefore, I suggest leaving the functionality how it is and labelling this issue <kbd>[wontfix](https://github.com/mybb/mybb/labels/wontfix)</kbd>.

**Comment by tommm** [comment]
My memory is a bit hazy on this subject, but we've discussed this internally before.

The ability to create usergroups with the same name is intentional. This is so you can have a group named _administrators_ with actual admin permissions, and a display group named _administrators_ that may or may not have any actual admin permissions. The ability for 'hidden' usergroups was deemed important in the early MyBB days.

A `flash_message` with `You have multiple usergroups named X. You may want to rename them to avoid confusion.` may be sufficient.

**Comment by Cu8eeeR** [comment]
@SvePu and others - I suggest to use a notification bar (message) when you wanna create a new group with the same name. It would be enough.

**Comment by Sama34** [comment]
I agree with @WildcardSearch 's suggestion for a fix.

**Comment by JordanMussi** [comment]
Implemented @tommm's suggestion in #2838.

**Comment by WildcardSearch** [comment]
Since this hasn't been pushed yet, I want to add that I think the current language is annoying. I've always hated software that used condescending language-- and in the context of an admin purposely naming two groups the same for some customization and being told that he "...might want to rename the group to avoid confusion..." would be frustrating.

Keep it simple, I say:

> The new user group has been created successfully. Notice: There are multiple user groups named \"{1}\".

**Comment by JordanMussi** [comment]
@WildcardSearch I support that, fixed in 85e640af00f8be38b5ac5e50fcfea3d7850c5727

**Comment by WildcardSearch** [comment]
Great @JordanMussi and @Shade- 😄
